examples/basics/visuals/tube.py

```diff
@@ -56,7 +56,7 @@
 
 l4 = scene.visuals.Tube(points4,
                         color=colors,
-                        shading='flat',
+                        shading='smooth',
```

Comment:
I thought I asked this before, but why the change in shading here?

Reply:
`flat` doesn't work with the mesh in `mode=lines`.
See the bottom-right tube in this image:
![tube_lines_shading](https://user-images.githubusercontent.com/830589/82141831-b0cc7000-9838-11ea-804e-55c765a2ef04.png)
Left: `shading=smooth`. Right: `shading=flat`.

I think this is because flat shading is implemented directly in [this way](https://github.com/vispy/vispy/blob/5002cb117810a476f8ce29240de849746d88bdc4/vispy/visuals/filters/mesh.py#L64). That is, the normals are computed in the shader, assuming that we are drawing faces (I suspect `dFdx` and `dFdy` do not make sense in `mode=lines`).
This was initially introduced to allow flat shading with indexed buffers (and limitations of the GLSL version, I think). However, at the moment, this is not needed since we assume expanded buffers of faces (where we could pass a single normal direction for each face instead). It might still be useful in the future but there are also other approaches, e.g. the `flat` [type qualifier](https://www.khronos.org/opengl/wiki/Type_Qualifier_(GLSL)) (search for `flat` in the link) but not in earlier GLSL, I think.
I'd have to double-check these claims in practice. If they hold, we could change the content of the normals buffer instead of the current proposed approach for `flat` shading to work in `mode=lines`. Do we absolutely need `flat` shading in `lines` mode if smooth works?

Reply:
It depends what worked before, but in the end I think whatever works is good. In the long term if there is an approach to all this that makes every shading style work with every Mesh then great. If that approach causes performance problems or months of development time then it isn't worth it (not for free at least).

I can't say what users will want to use this for so until someone requests it or adds the functionality I think we stick with what you've done.